tracepoint_perf/examples/rewrite_perf.rs

```diff
@@ -0,0 +1,280 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//! Demonstrates how to use [`tp::PerfDataFileWriter`] to write events to a
+//! `perf.data` file. In this example, the events are read from an existing
+//! `perf.data` file using [`tp::PerfDataFileReader`] and written to a new
+//! `perf.data` file using [`tp::PerfDataFileWriter`].
+//!
+//! This is not intended to be useful (except perhaps for testing purposes). It is
+//! intended to show how `PerfDataFileReader` can be used to take the `perf.data` file
+//! apart and how `PerfDataFileWriter` can put it back together.
+//!
+//! Note that the output file is not expected to be exactly the same as the input:
+//!
+//! - Output is always a normal-mode file even if input was a pipe-mode file.
+//! - Output file may store headers in a different order.
+//! - Output file may use more/less padding.
+//! - If the input file is semantically inconsistent, the output file may not
+//!   precisely match the input (the inconsistent data might be lost). For
+//!   example, there are usually two (or more) copies of each attr, one in a
+//!   v1 format and another in a v2 format. The rewrite process will typically
+//!   ignore the v1 copy of the data if a v2 copy is available, so if the v1 copy
+//!   is semantically different from the v2 copy, that detail might be lost during
+//!   rewrite.
+
+use std::collections;
+use std::env;
+use std::fs;
+use std::io;
+use std::process;
+use std::string;
+use std::vec;
+
+use tracepoint_decode as td;
+use tracepoint_perf as tp;
+
+fn main() -> process::ExitCode {
```

Comment:
Note: In my tests, this produces output identical to the output of LinuxTracepoints `tracepoint-perf-file-rewrite-sample`.